tests/Tests_RangeEnhancedGenerator.py

```diff
@@ -0,0 +1,48 @@
+
+# coding: utf-8
+
+# In[ ]:
+
+
+import unittest
+import RangeEnhancedIterator_Generator
+from RangeEnhancedIterator_Generator import GeneratorEnhancedRange
+
+
+class GeneratorTests(unittest.TestCase):
+
+    def setUp(self):
+        print ('Setup called')
+    
+    def test_on_range_w_one_arg(self):
+        example = GeneratorEnhancedRange(2)
+        s = []
+        for i in range(10):
+            for i in example():
+                s.append(i)
+        self.assertEqual(s, [0, 1])
+    
+    def test_on_range_w_two_args(self):
+        example = GeneratorEnhancedRange(2,5)
+        s = []
+        for i in range(10):
+            for i in example():
+                s.append(i)
+        self.assertEqual(s, [2, 3, 4])
```

Comment:
Could we simplify this into:
```
self.assertEqual([n for n in GeneratorEnhancedRange(2,5)], [2, 3, 4])
```
?

Reply:
I substituted my cycle for your code and the test failed. Your code returns [2], while my cycle results in [2, 3, 4] and passes the test.

Reply:
Update: after some modifications of the generators' code this test also works. Although I still don't know why.